bin/reth/src/jwt_secret_provider.rs

```diff
@@ -0,0 +1,134 @@
+use std::path::Path;
+
+use reth_primitives::hex;
+use reth_rpc::JwtSecret;
+use tracing::info;
+
+use crate::dirs::{JwtSecretPath, PlatformPath};
+
+type BoxError = Box<dyn std::error::Error>;
+
+/// The execution layer and consensus layer clients SHOULD accept a configuration parameter:
+/// jwt-secret, which designates a file containing the hex-encoded 256 bit secret key to be used
+/// for verifying/generating JWT tokens.
+///
+/// If such a parameter is given, but the file cannot be read, or does not contain a hex-encoded
+/// key of 256 bits, the client SHOULD treat this as an error.
+///
+/// If such a parameter is not given, the client SHOULD generate such a token, valid for the
+/// duration of the execution, and SHOULD store the hex-encoded secret as a jwt.hex file on
+/// the filesystem. This file can then be used to provision the counterpart client.
+pub(crate) struct JwtSecretProvider(JwtSecret);
+
+impl JwtSecretProvider {
```

Comment:
This doesn't seem to be doing much in addition to `JwtSecret`? Let's just move these methods into ` JwtSecret` and use that everywhere?

Reply:
Probably a matter of taste.
In general boot logics should be kept at the edge of the application, but your approach is cleaner:
* Jwt bytes are kept private
* Jwt logics are not spread across the project

I'll move this as suggested